packages/utilities/src/memoize.ts

```diff
@@ -32,15 +32,18 @@ export function setMemoizeWeakMap(weakMap: any): void {
 export function memoize<T extends Function>(
   target: any,
   key: string,
-  descriptor: TypedPropertyDescriptor<T>) {
+  descriptor: TypedPropertyDescriptor<T>): {
+    configurable: boolean;
+    get(): any;
+  } {
 
   // We bind to "null" to prevent people from inadvertently pulling values from "this",
   // rather than passing them in as input values which can be memoized.
   let fn = memoizeFunction(descriptor.value && descriptor.value.bind(null));
 
   return {
     configurable: true,
-    get() {
+    get(): any {
```

Comment:
If TSLint is going to enforce `typedef` here that sort of defeats the point of this syntax.

Reply:
Should be `T`.